pkg/resmap/resmap.go

```diff
@@ -222,11 +222,13 @@ func MergeWithOverride(maps ...ResMap) (ResMap, error) {
 					glog.V(4).Infof("Replace %v with %v", result[id].Object, r.Object)
 					r.Replace(result[id])
 					result[id] = r
+					result[id].ChangeBehavior(resource.BehaviorCreate)
 				case resource.BehaviorMerge:
 					glog.V(4).Infof("Merging %v with %v", result[id].Object, r.Object)
 					r.Merge(result[id])
 					result[id] = r
 					glog.V(4).Infof("Merged object is %v", result[id].Object)
+					result[id].ChangeBehavior(resource.BehaviorCreate)
```

Comment:
I realize that the test data above captures the use of the overlay/base, but wondering if you could also capture this in the existing test of MergeWithoutOverride in resmap_test.go
(or make a new test).

E.g. confirm that the resource behavior field gets reset.

Reply:
BTW, that test is called TestMerge, my mistake, please fix :)

Reply:
I changed it to `TestMergeWithoutOverride` and added another function `TestMergeWithOverride`.